### PR TITLE
Added support for integrating the package_version

### PR DESCRIPTION
**Pull request by hitbear**
The package version information appears in a comment of the header file. The new command line argument for this is --package-version. The TOML config file flag is package_version=true||false.

This new feature is helpful when you are working on a growing Rust library with new functions and structs. The package version entry in the header file appears as
**"/*Package version: {}*/",**

and gives a better overview which header file fits to which version of the library.

### CODE REVIEW

**Review by emilio — CHANGES_REQUESTED** [review]
Looks fine, but needs a test, and probably undoing the rust-toolchain change. Let me know if you need any help with that.

File: rust-toolchain.toml [hunk 1]
```diff
@@ -1,2 +0,0 @@
-[toolchain]
-channel = "nightly"
```
[the remark this answers is not in the document]
Comment (hitbear) [reply rust-toolchain.toml:2]: Thank you for your response. Yes I fixed both requests but I think that I need some help to write an appropriate test.

[the remark this answers is not in the document]
Comment (emilio) [reply rust-toolchain.toml:2]: You can add files to `tests/rust` where your toml file has the new option.

[the remark this answers is not in the document]
Comment (hitbear) [reply rust-toolchain.toml:2]: Ok, in my newest commit I added a test crate package_version that checks whether the file is generated with the correct package version. 
Since some other tests failed with cython, I modified cbindgen. Now Comments such as header, trailer, etc. are written to the file after a check if the comment should be in C/C++ syntax (/*) or Python syntax ('''). 

`cargo test` now tells `ok. 145 passed;`

File: src/bindgen/bindings.rs [hunk 2]
```diff
@@ -209,9 +212,30 @@ impl Bindings {
             return;
         }
 
+        if self.config.package_version {
+            out.new_line_if_not_start();
+            match self.config.language {
+                Language::C | Language::Cxx => {
+                    write!(out, "/* Package version: {} */", self.package_version,);
```
Comment (emilio) [inline src/bindgen/bindings.rs:219]: I'd remove the trailing comma here and below.

File: src/bindgen/bindings.rs [hunk 3]
```diff
@@ -228,16 +252,35 @@ impl Bindings {
...
+                    );
+                }
+            }
+
             out.new_line();
         }
         if let Some(ref f) = self.config.autogen_warning {
             out.new_line_if_not_start();
-            write!(out, "{}", f);
+            match self.config.language {
```
Comment (emilio) [inline src/bindgen/bindings.rs:276]: I think we should keep these as it was, this contents are controllable by the user so we shouldn't try to mangle them.

File: src/bindgen/bindings.rs [hunk 4]
```diff
@@ -209,9 +212,30 @@ impl Bindings {
...
+                }
+                Language::Cython => {
+                    write!(out, "''' Package version: {} '''", self.package_version,);
+                }
+            }
+
+            out.new_line();
+        }
+
         if let Some(ref f) = self.config.header {
```
Comment (emilio) [inline src/bindgen/bindings.rs:229]: Same here, this should probably remain as it was.

File: src/bindgen/bindings.rs [hunk 5]
```diff
@@ -228,16 +252,35 @@ impl Bindings {
...
+                    write!(
+                        out,
+                        "/* Generated with cbindgen:{} */",
+                        crate::bindgen::config::VERSION
+                    );
+                }
+                Language::Cython => {
+                    write!(
+                        out,
+                        "''' Generated with cbindgen:{} '''",
```
Comment (emilio) [inline src/bindgen/bindings.rs:266]: This fix is obviously good.

**Review by emilio — CHANGES_REQUESTED** [review]
Thanks! I think we should keep user inputs as-is and not mangle them, but the rest looks fine, thank you!

File: src/bindgen/bindings.rs (hunk 4 above)
Comment (hitbear) [reply src/bindgen/bindings.rs:229]: So with these two commits, I removed the user input modification - and adjusted also the package_version test. Yes I see, it could be dangerous to change input.

File: tests/rust/package_version/cbindgen.toml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package_version = true
+
+[parse.expand]
```
Comment (emilio) [inline tests/rust/package_version/cbindgen.toml:3]: Do you need to use expand for this to work?

File: tests/rust/package_version/cbindgen.toml [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+crates = ["package_version"]
+features = ["cbindgen"]
+
```
Comment (emilio) [inline tests/rust/package_version/cbindgen.toml:6]: remove redundant newlines?

File: tests/rust/package_version/src/lib.rs [hunk 8]
```diff
@@ -0,0 +1,8 @@
+#[repr(C)]
+pub struct Foo {
+    #[cfg(not(feature = "cbindgen"))]
```
Comment (emilio) [inline tests/rust/package_version/src/lib.rs:3]: This seems orthogonal to the feature you're testing?

Comment (hitbear) [reply tests/rust/package_version/src/lib.rs:3]: Yes, actually I could remove the whole stub code but in my latest commit, i removed just this line.

**Review by emilio — APPROVED** [review]
